homeassistant/components/sensibo/climate.py

```diff
@@ -176,9 +176,7 @@ async def async_set_temperature(self, **kwargs) -> None:
             raise HomeAssistantError(
                 "Current mode doesn't support setting Target Temperature"
             )
-
-        if (temperature := kwargs.get(ATTR_TEMPERATURE)) is None:
-            return
```

Comment:
This check is needed, because the following service call will pass validation but does not contain temperature:
```yaml
service: climate.set_temperature
data:
  hvac_mode: 'off'
  target_temp_high: 24
  target_temp_low: 22
target:
  device_id: redacted
```

I would even say we can consider raise `HomeAssistantError` in this case

Reply:
Right. I was struggling with finding a case where temp could be None but you found one of course.
I will add it into the tests instead and revert the code